src/material/stepper/step-header.html

```diff
@@ -10,6 +10,12 @@
       [ngTemplateOutletContext]="_getIconContext()"></ng-container>
     <ng-container *ngSwitchDefault [ngSwitch]="state">
       <span *ngSwitchCase="'number'">{{_getDefaultTextForState(state)}}</span>
+      <span class="cdk-visually-hidden" *ngIf="state === 'done'">
+        {{_intl.completedLabel}}
+      </span>
+      <span class="cdk-visually-hidden" *ngIf="state === 'edit'">
+        {{_intl.editableLabel}}
+      </span>
```

Comment:
One thing that also occurred to me: we should compare doing this (inline text) vs. `aria-describedby` on the `role="tab"` element and see if one is preferable.

Reply:
When I tried using the `aria-describedby` attribute, nothing was announced when testing with ChromeVox so I just left it as is. I also added tests